### ISSUE TITLE
Unhandled Rejection (Error): when oauth flow return server error

### ISSUE THREAD

**Issue by hp0210** [aws-amplify/amplify-js#3932]
**Describe the bug**
We are currently on the process of upgrading our aws-amplify library from 1.1.9 to 1.1.36 and aws-amplify-react from 1.0.19 to 2.3.12 . We found the following issue where we will get **Unhandled Rejection (Error)** error from aws-amplify library when our user is redirected back to the app with OAuth flow error (e.g https://localhost:3000/?error_description=Error+in+SAML+response&error=server_error)

**Additional context:**
* aws-amplify version 1.1.36
* aws-amplify-react version 2.3.12
* We use Cognito preSignUp trigger to make sure that only federated users that have been provisioned can access our application. So with any federated user that logs in via SSO flow (SAML provider via Cognito hosted UI) that doesn't have an account in our system, we will throw an exception in Cognito preSignUp trigger to prevent the user logging in the system. This worked fine with before we upgrading the library

**To Reproduce**
Steps to reproduce the behaviour:
1. Configure your aws-amplify with oauth config
2. Navigate to your app with error params (e.g https://localhost:3000/?error_description=Error+in+SAML+response&error=server_error) 
3. You will see the Unhandled Rejection (Error)

**Expected behaviour**
React app will be freezed due to Unhandled Rejection (Error)

**Our code**
```javascript
let amplifyConfig = {
  Auth: {
    identityPoolId: Config.cognito.IDENTITY_POOL_ID,
    region: Config.cognito.REGION,
    userPoolId: Config.cognito.USER_POOL_ID,
    userPoolWebClientId: Config.cognito.APP_CLIENT_ID,
    mandatorySignIn: true,
    cookieStorage: {
      domain: Config.reactApp.COOKIE_DOMAIN,
      path: '/',
      expires: 1,
      secure: Config.reactApp.SECURE_PROTOCOL
    },
    oauth: {
      domain : Config.cognito.USERPOOL_HOSTED_DOMAIN+'.auth.'+Config.cognito.REGION+'.amazoncognito.com',
      scope : [ 'openid', 'email' ],
      redirectSignIn : Config.cognito.USERPOOL_ADFS_CALLBACK_URL,
      redirectSignOut : Config.cognito.USERPOOL_ADFS_LOGOUT_URL,
      responseType: 'code',
    }
  },
  Analytics: {
    disabled: true,
    autoSessionRecord: false,
  }
};

Amplify.configure(amplifyConfig)
```
**Screenshots**
<img width="1918" alt="Screen Shot 2019-08-28 at 11 45 47 am" src="https://user-images.githubusercontent.com/3418775/63819778-edde9280-c989-11e9-87ac-051de6f93a26.png">
<img width="1920" alt="Screen Shot 2019-08-28 at 11 45 58 am" src="https://user-images.githubusercontent.com/3418775/63819781-f040ec80-c989-11e9-8e3f-f9c8fb93e2e7.png">
<img width="1920" alt="Screen Shot 2019-08-28 at 11 46 31 am" src="https://user-images.githubusercontent.com/3418775/63819786-f20ab000-c989-11e9-9802-85160ece0c2d.png">

**Comment by rabellino12** [comment]
I'm having the same issue, there is no trycatch or componentDidCatch that helps to prevent the app to freeze

**Comment by hp0210** [comment]
@rabellino12 please +1 for this issue.
However, this issue doesn't affect our production build as in production environment the error won't frezee the entire stack. It does affect our local development though

**Comment by rabellino12** [comment]
+1

**Comment by haverchuck** [comment]
@rabellino12 @hp0210 We are looking into this.

**Comment by haverchuck** [comment]
@rabellino12 @hp0210 - Just to be clear, the expected behavior here is for the application to load in an unauthenticated state (with the ideal case also being that the error from the url bubbles up through Amplify's standard error logging system).  Is that correct?

**Comment by rabellino12** [comment]
Hi @haverchuck thanks for looking into it, yes, so we can handle custom rejections sent from pre signup trigger and give some feedback to the user about why the authentication/signup failed

**Comment by haverchuck** [comment]
@rabellino12 @hp0210  -  I've put in [a PR to fix this](https://github.com/aws-amplify/amplify-js/pull/3993) (a simple try/catch).  It will probably be a bit before this winds its way to our release branch, but calling it out in case you want to take a look to unblock yourself.

**Comment by rabellino12** [comment]
Great, thank you @haverchuck

**Comment by haverchuck** [comment]
@rabellino12 @rabellino12  - PR for this has been merged into our master branch, which means it should be available in npm under the '@unstable' tag as soon as it makes its way through our pipeline.  We will get a release to the main build as soon as we can.
